scripts/python/migrate_data_2024.py

```diff
@@ -0,0 +1,2683 @@
+            cursor.execute(sql_query)
+            data = cursor.fetchall()
+            if len(data) != 0:
+                for row in data:
+                    attribs_data[row[0]] = { 'attrib_value':row[1],
+                                            'attrib_type_code':row[2],
+                                            'attrib_type_name':row[3],
+                                            'attrib_type_description':row[4]}
+
+    except Error as e:
+        print("Error while connecting to MySQL", e)
+    finally:
+        if connection.is_connected():
+            cursor.close()
+            connection.close()
+
+    return attribs_data
+
+"""
+    Return the panel name and the visibility
+"""
+def dump_panels(host, port, db, user, password):
+    panels_data = {}
+
+    map_name = {
+        "DD": "Developmental disorders",
+        "Ear": "Ear disorders",
+        "Eye": "Eye disorders",
+        "Skin": "Skin disorders",
+        "Cancer": "Cancer disorders",
+        "Prenatal": "Prenatal disorders",
+        "Neonatal": "Neonatal disorders",
+        "Demo": "Demo",
+        "Rapid_PICU_NICU": "Rapid_PICU_NICU disorders",
+        "PaedNeuro": "PaedNeuro disorders",
+        "Skeletal": "Skeletal disorders",
+        "Cardiac": "Cardiac disorders",
+        "Hearing loss": "Hearing loss"
+    }
+
+    sql_query = f""" SELECT name, is_visible
+                     FROM panel """
+    
+    connection = mysql.connector.connect(host=host,
+                                         database=db,
+                                         user=user,
+                                         port=port,
+                                         password=password)
+
+    try:
+        if connection.is_connected():
+            cursor = connection.cursor()
+            cursor.execute(sql_query)
+            data = cursor.fetchall()
+            if len(data) != 0:
+                for row in data:
+                    panels_data[row[0]] = { 'description': map_name[row[0]], 'is_visible':row[1] }
+
+    except Error as e:
+        print("Error while connecting to MySQL", e)
+    finally:
+        if connection.is_connected():
+            cursor.close()
+            connection.close()
+
+    return panels_data
+
+"""
+    Return the users data and the panels they can curate
+"""
+def dump_users(host, port, db, user, password, attribs):
+    users_data = {}
+
+    sql_query_user = f""" SELECT username, email, panel_attrib
+                          FROM user """
+    
+    connection = mysql.connector.connect(host=host,
+                                         database=db,
+                                         user=user,
+                                         port=port,
+                                         password=password)
+
+    try:
+        if connection.is_connected():
+            cursor = connection.cursor()
+            cursor.execute(sql_query_user)
+            data = cursor.fetchall()
+            if len(data) != 0:
```

Comment:
Instead of` if len(data) != 0:`  I am suggesting we use` if data `
Applies to other lines too